File: app/src/main/java/com/example/david/lists/view/common/ListViewLogicBase.java
package com.example.david.lists.view.common;

import com.example.david.lists.data.repository.IRepositoryContract;
import com.example.david.lists.util.ISchedulerProviderContract;

import io.reactivex.disposables.CompositeDisposable;

public abstract class ListViewLogicBase {

    protected final IRepositoryContract.Repository repo;
    protected final ISchedulerProviderContract schedulerProvider;
    protected final CompositeDisposable disposable;

    protected ListViewLogicBase(IRepositoryContract.Repository repo,
                                ISchedulerProviderContract schedulerProvider,
                                CompositeDisposable disposable) {
        this.repo = repo;
        this.schedulerProvider = schedulerProvider;
        this.disposable = disposable;
    }
}

File: app/src/main/java/com/example/david/lists/view/common/ListViewLogicBase.kt
package com.example.david.lists.view.common

import com.example.david.lists.util.ISchedulerProviderContract
import com.example.domain.repository.IRepositoryContract
import io.reactivex.disposables.CompositeDisposable

abstract class ListViewLogicBase protected constructor(protected val repo: IRepositoryContract.Repository,
                                                       protected val schedulerProvider: ISchedulerProviderContract,
                                                       protected val disposable: CompositeDisposable)
